line.go

```diff
@@ -104,7 +105,7 @@ func (s *State) refreshSingleLine(prompt []rune, buf []rune, pos int) error {
 		return err
 	}
 
-	pLen := countGlyphs(prompt)
+	pLen := countGlyphs([]rune(regexp.MustCompile("[\u001b\u009b][[()#;?]*(?:[0-9]{1,4}(?:;[0-9]{0,4})*)?[0-9A-ORZcf-nqry=><]").ReplaceAllString(string(prompt), ""))) // Remove ANSI escape codes, at least most of them, regex took from https://github.com/chalk/ansi-regex/blob/master/index.js#L3
```

Comment:
Please consider adding the comment above this line and break the expression into multiple so as to not cross 80 chars.